source/ota.c

```diff
@@ -1817,8 +1817,20 @@ static DocParseErr_t parseJSONbyModel( const char * pJson,
     uint16_t paramIndex = 0;
     const char * pFileParams = NULL;
     uint32_t fileParamsLength = 0;
+    static char cJsonBuffer[ OTA_MAX_JSON_STR_LEN ];
```

Comment:
Why not make this `cJsonBuffer[ OTA_MAX_JSON_STR_LEN + 1]`? Then you can store the maximum message length guaranteed.

Reply:
You could also get rid of the if/else here assuming `OTA_MAX_JSON_STR_LEN` by doing this

Reply:
OTA_MAX_JSON_STR_LEN(256) is not the maximum size of whole json file. So that if/else case must be there.
Here is the example output from my test. There are about 600 charactors.
```
[DEBUG] [OTA] [ota.c:1761] JSON received: {"timestamp":1690449009,"execution":{"jobId":"AFR_OTA-
tryOTA091aaa","status":"QUEUED","queuedAt":1690449008,"lastUpdatedAt":1690449008,"versionNumber":1,
"executionNumber":1,"jobDocument":{"afr_ota":{"protocols":["MQTT"],"streamname":
"AFR_OTA-b16fc240-3c43-4778-a08c-5bf691ede395","files":[{"filepath":
"/home/ousc/git/aws-iot-device-sdk-embedded-C/build/bin/0911",
"filesize":236048,"fileid":0,"certfile":"certificates/ecdsasigner.crt","sig-sha256-ecdsa":
"MEUCIQD1PBg3xq6GmknDymDa7qXDc2gg7LvU/fk5sZnCevPacwIgJB5f5F1py8HdCo7FQdUQNCQEgv1o/C7vwqIv2+QBE2s="}]}}}}
```